controllers/nodemaintenance_controller.go

```diff
@@ -55,16 +72,379 @@ type NodeMaintenanceReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+	r.logger = logger
```

Comment:
I think using both a logger per scope and mutating the inner logger is misleading.
maybe `r.logger = log.withFields...` and then only `r.logger` in all places?

Reply:
fair point, but I go with `r.logger = log.FromContext(ctx)`